resources/js/group-history/search-form.tsx

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+import BigButton from 'components/big-button';
+import InputContainer from 'components/input-container';
+import { action, makeObservable } from 'mobx';
+import { observer } from 'mobx-react';
+import * as React from 'react';
+import { trans } from 'utils/lang';
+import groupStore from './group-store';
+import GroupHistoryJson from './json';
+
+const bn = 'group-history-search-form';
+const formParamKeys = ['group', 'max_date', 'min_date', 'user'] as const;
+
+interface Props {
+  currentParams: GroupHistoryJson['params'];
+  loading: boolean;
+  newParams: GroupHistoryJson['params'];
+  onSearch: () => void;
+}
+
+@observer
+export default class SearchForm extends React.Component<Props> {
+  constructor(props: Props) {
+    super(props);
+    makeObservable(this);
+  }
+
+  render() {
+    const newParamsEmpty = formParamKeys.every(
+      (key) => this.props.newParams[key] == null,
+    );
+    const newParamsSame = formParamKeys.every(
+      (key) => this.props.newParams[key] === this.props.currentParams[key],
+    );
+
+    return (
+      <form className={bn} data-loading-overlay='0' onSubmit={this.onSubmit}>
+        <div className={`${bn}__content ${bn}__content--inputs`}>
+          <InputContainer for={`${bn}-group`} labelKey='group_history.form.group'>
+            <div className='form-select form-select--group-history'>
+              <select
+                className='form-select__input'
+                id={`${bn}-group`}
+                name='group'
```

Comment:
`for`, `name`, and `id` aren't needed?

Reply:
`name` is for `onChange()` 

and hm I thought `for`/`id` would give the same kind of click target behavior as on the other inputs but i guess not. I've seen this used on label+select before, does it help with accessibility or something? :thinking:

Reply:
it's not needed when the input is inside the label.